cli/cmd/generate_oci.go

```diff
@@ -0,0 +1,446 @@
+		return errors.Wrap(err, "collecting/confirming parameters")
+	}
+
+	return nil
+}
+
+func initGenerateOciTfCommandFlags() {
+	// add flags to sub commands
+	generateOciTfCommand.PersistentFlags().BoolVar(
+		&GenerateOciCommandState.Config,
+		"config",
+		false,
+		"enable config integration")
+	generateOciTfCommand.PersistentFlags().StringVar(
+		&GenerateOciCommandState.ConfigName,
+		"config_name",
+		"",
+		"specify name of config integration")
+	generateOciTfCommand.PersistentFlags().StringVar(
+		&GenerateOciCommandState.TenantOcid,
+		"tenant_ocid",
+		"",
+		"specify the OCID of the tenant to integrate")
+	generateOciTfCommand.PersistentFlags().StringVar(
+		&GenerateOciCommandState.OciUserEmail,
+		"oci_user_email",
+		"",
+		"specify the email of the OCI user created for integration")
+	generateOciTfCommand.PersistentFlags().BoolVar(
+		&GenerateOciCommandExtraState.TerraformApply,
+		"apply",
+		false,
+		"run terraform apply without executing plan or prompting",
+	)
+	generateOciTfCommand.PersistentFlags().StringVar(
+		&GenerateOciCommandExtraState.Output,
+		"output",
+		"",
+		"location to write generated content (default is ~/lacework/oci)",
+	)
+}
+
+// basic validation of Tenant OCID format
+func validateOciTenantOcid(val interface{}) error {
+	return validateStringWithRegex(
+		val,
+		`ocid1\.tenancy\.[^\.\s]*\.[^\.\s]*(\.[^\.\s]+)?\.[^\.\s]+`,
+		"invalid tenant OCID supplied",
+	)
+}
+
+// basic validation of email
+func validateOciUserEmail(val interface{}) error {
+	return validateEmailAddress(val)
+}
+
+func promptCustomizeOciOutputLocation(extraState *OciGenerateCommandExtraState) error {
+	if err := SurveyQuestionInteractiveOnly(SurveyQuestionWithValidationArgs{
+		Prompt:   &survey.Input{Message: QuestionOciCustomizeOutputLocation, Default: extraState.Output},
+		Response: &extraState.Output,
+		Opts:     []survey.AskOpt{survey.WithValidator(validPathExists)},
+		Required: true,
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func promptCustomizeOciConfigOptions(config *oci.GenerateOciTfConfigurationArgs) error {
+	if err := SurveyQuestionInteractiveOnly(SurveyQuestionWithValidationArgs{
+		Prompt:   &survey.Input{Message: QuestionOciConfigName, Default: config.ConfigName},
+		Checks:   []*bool{&config.Config},
+		Response: &config.ConfigName,
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func askAdvancedOciOptions(config *oci.GenerateOciTfConfigurationArgs, extraState *OciGenerateCommandExtraState) error {
+	answer := ""
+
+	// Prompt for options
+	for answer != OciAdvancedOptDone {
+		// Construction of this slice is a bit strange at first look, but the reason for that is because
+		// we have to do string validation to know which option was selected due to how survey works; and
+		// doing it by index (also supported) is difficult when the options are dynamic (which they are)
+		//
+		// Only ask about more accounts if consolidated cloudtrail is setup (matching scenarios doc)
```

Comment:
Not sure if this comments still makes sense?

Reply:
Not at all. I'll remove it.